tests/target/macros/matches/version_two.rs

```diff
@@ -0,0 +1,154 @@
+//rustfmt-version: Two
+
+// To visually verify the special case handling of `matches!` we include the equivalent match expr
+
+// issue #4462
+fn issue_4462() {
+    matches!(
+        c,
+        'x' | 'c' | 'b' | 'B' | '?' | 'h' | 'H' | 'i' | 'I' | 'l' | 'L' | 'q' | 'Q' | 'n' | 'N'
+        | 'f' | 'd' | 's' | 'p' | 'P'
+    );
+
+    match c {
+        'x' | 'c' | 'b' | 'B' | '?' | 'h' | 'H' | 'i' | 'I' | 'l' | 'L' | 'q' | 'Q' | 'n' | 'N'
+        | 'f' | 'd' | 's' | 'p' | 'P' => {}
+    }
+}
+
+// issue #4885
+fn issue_4885() {
+    matches!(
+        c,
+        '\\' | '.' | '+' | '(' | ')' | '|' | '[' | ']' | '{' | '}' | '^' | '$' | '#' | '&' | '-'
+        | '~' | '*' | '?'
+    );
+
+    match c {
+        '\\' | '.' | '+' | '(' | ')' | '|' | '[' | ']' | '{' | '}' | '^' | '$' | '#' | '&'
+        | '-' | '~' | '*' | '?' => {}
+    }
+}
+
+// issue #5176
+fn issue_5176() {
+    matches!(self, Self::A | Self::B);
+    match self {
+        Self::A | Self::B => {}
+    }
+}
+
+// issue #5547
+fn issue_5547() {
+    matches!(
+        something.very_very_very.long.even.more.fields,
+        Some(very_long_field_name_name_to_check) if method(very_long_field_name)
+    );
+    match something.very_very_very.long.even.more.fields {
+        Some(very_long_field_name_name_to_check) if method(very_long_field_name) => {}
+    }
+}
+
+// other scenarios
+fn other_scenarios() {
+    // no guard with trailing comma
+    matches!(self, Self::A | Self::B,);
+    match self {
+        Self::A | Self::B => {}
+    }
+
+    // guard with trailing comma
+    matches!(
+        something.very_very_very.long.even.more.fields,
+        Some(very_long_field_name_name_to_check) if method(very_long_field_name),
+    );
+    match something.very_very_very.long.even.more.fields {
+        Some(very_long_field_name_name_to_check) if method(very_long_field_name) => {}
+    }
+
+    // short expr and pattern, but guard is long.
+    matches!(
+        something,
+        Some(_)
+            if method(
```

Comment:
The indentation here looks weird in my opinion. The indentation makes sense in `match` context because a match might contain multiple arms but the `matches!` macro does not.

Reply:
I appreciate the feedback. My goal with this PR was to make `matches!` formatting consistent with `match` expression formatting, and this particular formatting is a result of reusing `rewrite_guard` to not have to reimplement guard formatting.

Reply:
@calebcartwright maybe `t-style` would want to chime in on the guard formatting here

Reply:
Has this been discussed somewhere or yet. If so, what was the conclusion?